metricbeat/autodiscover/builder/hints/metrics_test.go

```diff
@@ -67,6 +67,21 @@ func TestGenerateHints(t *testing.T) {
 			len:    0,
 			result: common.MapStr{},
 		},
+		{
+			message: "Hints without matching port should return nothing",
+			event: bus.Event{
+				"host": "1.2.3.4",
+				"port": 9090,
```

Comment:
What if no port is assigned to this container?

Reply:
Then a configuration is returned, the following test cases cover this.